scheds/rust/scx_chaos/src/bpf/main.bpf.c

```diff
@@ -79,13 +79,16 @@ static __always_inline enum chaos_trait_kind choose_chaos(struct chaos_task_ctx
 	if (taskc->match & CHAOS_MATCH_EXCLUDED)
 		return CHAOS_TRAIT_NONE;
 
-	if (bpf_get_prandom_u32() < random_delays_freq_frac32)
-		return CHAOS_TRAIT_RANDOM_DELAYS;
+	u32 roll = bpf_get_prandom_u32();
 
-	if (bpf_get_prandom_u32() < cpu_freq_frac32)
-		return CHAOS_TRAIT_CPU_FREQ;
+	#pragma unroll
+	for (int i = 0; i < CHAOS_TRAIT_MAX; ++i) {
+		if (roll <= trait_delay_freq_frac32[i])
```

Comment:
Long term it may be interesting to think about enabling multiple chaos traits at a time. If multiple traits have high frequencies it will be biased towards the trait that comes earlier in the iteration.